### PR TITLE
[TEVA-1502] Add jobseekers navigation items

### PR DESCRIPTION
**Pull request by cesidio**
## Jira tickets URL
https://dfedigital.atlassian.net/browse/TEVA-1502
https://dfedigital.atlassian.net/browse/TEVA-1564

## Figma
[Sprint-66](https://www.figma.com/file/IB5MUgCC5Kf084XxafUjG1/Sprint-66?node-id=2%3A110)

### Screen 0.0
Screenshot 2020-11-26 at 16 49 09

### Screen 1.4
Screenshot 2020-11-26 at 16 48 44

## Screenshots of UI changes:
Screenshot 2020-11-25 at 17 20 35
Screenshot 2020-11-26 at 16 44 04

### CODE REVIEW

**Comment by david-mears-dfe** [comment]
Like the current sign in box, this disappears at the 525px breakpoint. We haven't cared about that up till now because hiring staff are the only ones who need to sign in and they tend to be desktop users. But jobseekers will be on mobile.

Could you check UX's intentions for the breakpoint?

**Comment by cesidio** [comment]
> Like the current sign in box, this disappears at the 525px breakpoint. We haven't cared about that up till now because hiring staff are the only ones who need to sign in and they tend to be desktop users. But jobseekers will be on mobile.
> 
> Could you check UX's intentions for the breakpoint?

This ticket is not finished yet, we need to move the navigation items on the right. Was thinking to assign it to @alexbowen now that is mostly frontend, I think he can address that problem too

File: app/views/pages/home/_signin_jobseekers.html.haml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+%h2.govuk-heading-m= t('.title')
+
+%p
+  = t('.description')
+  = link_to t('.signup_link'), new_jobseeker_registration_path, class: 'govuk-link govuk-link--no-visited-state'
```
Comment (cpjmcquillan) [inline app/views/pages/home/_signin_jobseekers.html.haml:5]: `govuk_link_to`

File: app/views/shared/_navigation.haml [hunk 2]
```diff
@@ -13,4 +13,12 @@
...
         = t('footer.for_schools')
       - unless ReadOnlyFeature.enabled?
-        %li{ class: active_link_class(new_identifications_path) }= link_to t('nav.sign_in'), new_identifications_path, class: 'govuk-header__link mobile-header-underline-text'
+        - if JobseekerAccountsFeature.enabled?
+          - if jobseeker_signed_in?
+            %li{ class: active_link_class(jobseekers_saved_jobs_path) }= link_to t('footer.your_account'), jobseekers_saved_jobs_path, class: 'govuk-header__link mobile-header-underline-text'
+            %li{ class: active_link_class(destroy_jobseeker_session_path) }= link_to t('nav.sign_out'), destroy_jobseeker_session_path, method: :delete, class: 'govuk-header__link mobile-header-underline-text'
+          - else
+            %li{ class: active_link_class(new_jobseeker_session_path) }= link_to t('buttons.sign_in'), new_jobseeker_session_path, class: 'govuk-header__link mobile-header-underline-text'
+            %li{ class: active_link_class(new_identifications_path) }= link_to t('nav.for_schools'), new_identifications_path, class: 'govuk-header__link mobile-header-underline-text'
```
Comment (cpjmcquillan) [inline app/views/shared/_navigation.haml:22]: For some reason the active class isn't applied here

**Review by cpjmcquillan — APPROVED** [review]
The more logic is added to the navbar, the more it might be better to create a navbar component - what do you think?

File: app/views/pages/home/_signin_jobseekers.html.haml (hunk 1 above)
Comment (cesidio) [reply app/views/pages/home/_signin_jobseekers.html.haml:5]: let's do all these changes in another PR

File: app/views/shared/_navigation.haml (hunk 2 above)
Comment (cesidio) [reply app/views/shared/_navigation.haml:22]: I have checked and it works for me, are you talking about the active class for `for_schools` item?

File: app/views/pages/home/_signin_jobseekers.html.haml (hunk 1 above)
Comment (cpjmcquillan) [reply app/views/pages/home/_signin_jobseekers.html.haml:5]: Ok, no problem

File: app/components/shared/navbar_component/navbar_component.scss [hunk 3]
```diff
@@ -0,0 +1,23 @@
+nav ul .mobile-header-underline-text {
+  @include govuk-media-query($until: desktop) {
+    font-weight: normal;
+    text-decoration: underline;
+  }
+}
+
+.mobile-header-top-border {
+  @include govuk-media-query($until: desktop) {
+    border-top: 3px solid #fff;
```
Comment (cpjmcquillan) [inline app/components/shared/navbar_component/navbar_component.scss:10]: Can we use the `govuk-colour`?

File: app/components/shared/navbar_component.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Shared::NavbarComponent < ViewComponent::Base
+  delegate :active_link_class, to: :helpers
+  delegate :current_organisation, to: :helpers
+  delegate :current_publisher_is_part_of_school_group?, to: :helpers
+  delegate :organisation_type_basic, to: :helpers
+  delegate :jobseeker_signed_in?, to: :helpers
+  delegate :publisher_signed_in?, to: :helpers
+
+  def initialize; end
```
Comment (csutter) [inline app/components/shared/navbar_component.rb:9]: Do we need an empty initializer here?

**Review by csutter — APPROVED** [review]
:shipit:

**Review by cpjmcquillan — APPROVED** [review]

File: app/components/shared/navbar_component/navbar_component.html.haml [hunk 5]
```diff
@@ -13,4 +13,12 @@
       %label.govuk-label.navigation_heading
```
Comment (alexbowen) [inline app/components/shared/navbar_component/navbar_component.html.haml:13]: nothing to do with this PR, but i am going to remove these labels. there is no point to them and you should only have li as direct children of ul/ol

Comment (alexbowen) [reply app/components/shared/navbar_component/navbar_component.html.haml:13]: ill comment in my subsequent PR

**Review by alexbowen — APPROVED** [review]

**Pull request merged**